src/libhtml/lex.c

```diff
@@ -734,15 +745,19 @@ getscriptdata(TokenSource* ts, int firstc, int starti, Token* a, int* pai)
 		if(c != 0){
 			buf[j++] = c;
 			if(j == BIGBUFSIZE-1){
+				tmpS = s;
```

Comment:
This pattern is repeated enough that something like,

```
static Rune*
Buftostr(Rune* s, Rune *buf, int j)
{
        Rune *tmp = s;
        s = buftostr(tmp, buf, j);
        free(tmp);
        return s;
}
```

Might be warranted.

Reply:
Yeah, I changed the buftostr API, and now the change is much nicer. (Every call was a memleak.)